python/pyspark/sql/catalog.py

```diff
@@ -132,6 +132,30 @@ def listFunctions(self, dbName=None):
                 isTemporary=jfunction.isTemporary()))
         return functions
 
+    def functionExists(self, functionName, dbName=None):
+        """Check if the function with the specified name exists.
+        This can either be a temporary function or a function.
+
+        .. versionadded:: 3.3.0
+
+        Parameters
+        ----------
+        functionName : str
+            name of the function to check existence
+        dbName : str, optional
+            name of the database to check function existence in.
+            If no database is specified, the current database is used
+
+        Returns
+        -------
+        bool
+            Indicating whether the function exists
+
```

Comment:
@dominikgehl since we're here, could we add some simple examples here as well?

Reply:
Sure. Done